middleware.go

```diff
@@ -142,6 +149,11 @@ func (tw traceware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	)
 	defer span.End()
 
+	// put trace_id to response header
+	if span.SpanContext().HasTraceID() {
+		w.Header().Add(tw.traceResponseHeaderKey, span.SpanContext().TraceID().String())
+	}
```

Comment:
By default, this will always add `X-Trace-ID` to the response header. However, this behavior might not be desirable for some people because, in the current implementation, we don't add `X-Trace-ID` to the response.

I think we should only add this header when the user only use `WithTraceResponseHeaderKey` option.